### PR TITLE
Integration of Login using our own backend service

### PR DESCRIPTION
**Pull request by Khukhuna**
**Part of GCI 2018** :tada: :balloon: 

## [Video Preview](https://www.youtube.com/watch?v=-wJCfKjN_gc)

Added authentication, saving of token & expiration date in **localStorage** on top of **PR 92**. (In case of merge It will not produce merge conflicts)

main changes for implementing authentications are in **auth.service.ts** & **login.component.ts**.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +22,9 @@
     "@angular/router": "^6.1.0",
     "ag-grid-angular": "^19.0.0",
     "ag-grid-community": "^19.0.0",
-    "bootstrap": "^4.1.3",
+    "bootstrap": "^3.3.7",
```
Comment (nikitadhiman) [inline package.json]: why are you using lower version of bootstrap?

File: src/app/core/auth.service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  baseUrl = "https://asidjaskldhaksudhauskd.herokuapp.com/";
```
Comment (nikitadhiman) [inline src/app/core/auth.service.ts]: we won't be hard coding base url

File: src/app/core/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  login(username:string, password: string){
+    let body = new URLSearchParams();
+    body.set('username', username);
+    body.set('password', password);
+
+    let options = {
+      headers: new HttpHeaders().set('Content-Type', 'application/x-www-form-urlencoded').set('Access-Control-Allow-Origin', '*')
+    };
+    return this.http.post(this.loginPath, body.toString(), options);
```
Comment (nikitadhiman) [inline src/app/core/auth.service.ts]: there a better ways to implement this, can you search and try them

**Comment by nikitadhiman** [comment]
Also, there are some unnecessary changes. Please remove them and include only changes/commits related to integration of backend services

File: src/app/core/auth.service.ts (hunk 2 above)
Comment (jatinkatyal13) [reply src/app/core/auth.service.ts]: We should get this URL from environment variables!

**Comment by nikitadhiman** [comment]
@Khukhuna , great job. This is being considered for merge, please remove  changes not related to integration of backend service {for eg footer,auth guard etc} so that we can go ahead and merger your awesome work into this project. Keep it up!

**Pull request closed without merging**